### DISCUSSION TITLE
Selling Partner API (SP API) : ApiError : Listing Items : putListingsItem with merchant_suggested_asin is on weird behavior

### DISCUSSION THREAD

**Discussion by qdev89** [amzn/selling-partner-api-models#573] [Ideas]
hi guys,
I'm really stuck on fileld merchant_suggested_asin, it's a required filed for a product type definitions (TEMPORARY_TATTOO) but I don't know how to fill it to complete my API call.
1. If I put 
` "merchant_suggested_asin": [
            {
                "value": "WPID||ASIN"
            }
        ],`
It will return as issue on 
` "issues": [
        {
            "code": "8105",
            "message": "The data you provided for an attribute is invalid. Please refer the data definitions and resubmit with a valid value.",
            "severity": "ERROR",
            "attributeNames": [
                "merchant_suggested_asin"
            ],
            "categories": [
                "INVALID_ATTRIBUTE"
            ]
        }
    ],`
    
    2. If I put a random number on this, it will return issue as :
    `"code": "8541",
            "message": "The Listing data provided is different from what's already in the Amazon catalog. The item_id provided matches ASIN 0000000000, but some of the listing data contradicts what is already in the Amazon catalog. The following listing attribute value(s) conflict with Amazon catalog value(s): 'item_name' (Merchant [\"100 PCS Two Fast Race Car Temporary Tattoos Stickers Theme 2nd Birthday Party Decorations Favors Supplies Decor 2 Fast Racing Checkered Tattoo Sticker Gifts For Boys Girls School Prizes Carnival\"] / Amazon [en_US: \"Finding Peaches in the Desert\", en_IN: \"Special Education Teachers Kit: The Cartography of Syntactic Structures, Volume 6\", en_GB: \"Ksiega Diny: The Cartography of Syntactic Structures, Volume 6 (Äldre Svenska Frälsesläkter)\" and 19 other variant(s) en_SG, en_CA, en_AE, en_AU, nl_NL, de_DE, fr_FR, pl_PL, es_ES, hi_IN, zh_CN, ja_JP, es_MX, fr_BE, pt_BR, sv_SE, it_IT, ar_AE, tr_TR]), 'customer_id' (Merchant [\"692323389502\"] / Amazon []). To fix the issue, please follow these steps 1) If the product you are listing is the same as 0000000000, then update your listing's 'item_name, customer_id' attribute(s) to match Amazon's catalog value, 2) If it's not the right ASIN, make sure the item_id provided is correct, or 3) If you disagree with the data in Amazon catalog contact Selling Partner Support to start the appeal process.",
            "severity": "ERROR",`
            
 3.  MY question is : how do I put it's as null because I want to listing a new listing that doesn't have exist item on it. 
            as scheme show as 
            
            ` "merchant_suggested_asin": {
            "title": "Merchant Suggested ASIN",
            "description": "Provide an ASIN for your product if one exists. If a value is not provided, the system will attempt a match based on the External Product ID.",
            "examples": [
                "B007KQBXN0"
            ],
            "type": "array",
            "minItems": 1,
            "minUniqueItems": 1,
            "maxUniqueItems": 1,
            "selectors": [
                "marketplace_id",
                "value"
            ],
            "items": {
                "type": "object",
                "required": [
                    "marketplace_id",
                    "value"
                ],
                "properties": {
                    "value": {
                        "title": "Merchant Suggested ASIN",
                        "description": "Provide an ASIN for your product if one exists. If a value is not provided, the system will attempt a match based on the External Product ID.",
                        "editable": false,
                        "hidden": false,
                        "examples": [
                            "B007KQBXN0"
                        ],
                        "type": "string",
                        "minLength": 10,
                        "maxLength": 10,
                        "maxUtf8ByteLength": 40
                    },
                    "marketplace_id": {
                        "$ref": "#/$defs/marketplace_id"
                    }
                },
                "additionalProperties": false
            }
        },`
        
        So it supposes can be empty but looks like it's a bug from SP API Listing.
        
        Please help me on this.
        Thank you

**Comment by weilinggu** [comment]
@qdev89 Have you checked the additional conditions in product type JSON schema? For example, in product type JSON schema (TEMPORARY_TATTOO)->`["allOf"][1]`, it indicated under what conditions would "merchant_suggested_asins" be required. I suggest take a look there (copied in this thread). Did you provide "`externally_assigned_product_identifier`" or "`supplier_declared_has_product_identifier_exemption`" property in your payload?

```
{
  "if": {
    "anyOf": [
      {
        "allOf": [
          {
            "not": {
              "required": [
                "externally_assigned_product_identifier"
              ],
              "properties": {
                "externally_assigned_product_identifier": {
                  "items": {
                    "required": [
                      "value"
                    ]
                  }
                }
              }
            }
          },
          {
            "not": {
              "required": [
                "parentage_level"
              ],
              "properties": {
                "parentage_level": {
                  "items": {
                    "required": [
                      "value"
                    ]
                  }
                }
              }
            }
          },
          {
            "required": [
              "supplier_declared_has_product_identifier_exemption"
            ],
            "properties": {
              "supplier_declared_has_product_identifier_exemption": {
                "contains": {
                  "required": [
                    "value"
                  ],
                  "properties": {
                    "value": {
                      "enum": [
                        false
                      ]
                    }
                  }
                }
              }
            }
          }
        ]
      },
      {
        "allOf": [
          {
            "not": {
              "required": [
                "externally_assigned_product_identifier"
              ],
              "properties": {
                "externally_assigned_product_identifier": {
                  "items": {
                    "required": [
                      "value"
                    ]
                  }
                }
              }
            }
          },
          {
            "not": {
              "required": [
                "parentage_level"
              ],
              "properties": {
                "parentage_level": {
                  "contains": {
                    "required": [
                      "value"
                    ],
                    "properties": {
                      "value": {
                        "enum": [
                          "parent"
                        ]
                      }
                    }
                  }
                }
              }
            }
          },
          {
            "required": [
              "supplier_declared_has_product_identifier_exemption"
            ],
            "properties": {
              "supplier_declared_has_product_identifier_exemption": {
                "contains": {
                  "required": [
                    "value"
                  ],
                  "properties": {
                    "value": {
                      "enum": [
                        false
                      ]
                    }
                  }
                }
              }
            }
          }
        ]
      },
      {
        "allOf": [
          {
            "not": {
              "required": [
                "externally_assigned_product_identifier"
              ],
              "properties": {
                "externally_assigned_product_identifier": {
                  "items": {
                    "required": [
                      "value"
                    ]
                  }
                }
              }
            }
          },
          {
            "not": {
              "required": [
                "parentage_level"
              ],
              "properties": {
                "parentage_level": {
                  "items": {
                    "required": [
                      "value"
                    ]
                  }
                }
              }
            }
          },
          {
            "not": {
              "required": [
                "supplier_declared_has_product_identifier_exemption"
              ],
              "properties": {
                "supplier_declared_has_product_identifier_exemption": {
                  "items": {
                    "required": [
                      "value"
                    ]
                  }
                }
              }
            }
          }
        ]
      },
      {
        "allOf": [
          {
            "not": {
              "required": [
                "externally_assigned_product_identifier"
              ],
              "properties": {
                "externally_assigned_product_identifier": {
                  "items": {
                    "required": [
                      "value"
                    ]
                  }
                }
              }
            }
          },
          {
            "not": {
              "required": [
                "parentage_level"
              ],
              "properties": {
                "parentage_level": {
                  "contains": {
                    "required": [
                      "value"
                    ],
                    "properties": {
                      "value": {
                        "enum": [
                          "parent"
                        ]
                      }
                    }
                  }
                }
              }
            }
          },
          {
            "not": {
              "required": [
                "supplier_declared_has_product_identifier_exemption"
              ],
              "properties": {
                "supplier_declared_has_product_identifier_exemption": {
                  "items": {
                    "required": [
                      "value"
                    ]
                  }
                }
              }
            }
          }
        ]
      }
    ]
  },
  "then": {
    "required": [
      "merchant_suggested_asin"
    ]
  }
}
```
If you are looking for some guidance understanding the product type JSON schema, try this video series https://www.youtube.com/watch?v=fraQT3UhFII on SP-API developer university
